pages/blog/graphql-deep-dive-1.mdx

```diff
@@ -0,0 +1,354 @@
+- Change the client implementation one by one to call the GraphQL gateway instead of the legacy service.
+- Now that you have GraphQL working in your ecosystem, you can incrementally move away from legacy implementations like SOAP without having to worry about how it will affect the clients progressively, one component at a time to use a native GraphQL implementation.
+
+While this is one possible approach, this is not the only approach. There are lot of other ways in which you can take this one step at a time without worrying about the legacy code you have.
+
+**How do I secure my GraphQL endpoint?**
+
+While the GraphQL spec itself does not recommend any specific way to do this and leaves it to the person implementing it, you can either use [**JWT**](https://jwt.io/)**,** Cookies, Sessions and so on like you normally would when authenticating through other mechanisms.
+
+**How do I enable authorization to my GraphQL fields or schema?**
+
+This is very powerful in GraphQL since you can do a authorization at a very fine grained level be it at the type level or at the field level. You can read [this](https://www.apollographql.com/blog/authorization-in-graphql-452b1c402a9/) blog which suggests various ways in which you can do authorization.
+
+You can also use libraries like [**GraphQL Shield**](https://github.com/maticzav/graphql-shield) which offers powerful middlewares to do this. But remember that authorization does come with attached cost since you are running a specific logic in/before your resolvers for all the fields which you want to authorize.
+
+One often overlooked way is the use of [**directives**](https://www.graphql-tools.com/docs/schema-directives/) to do authorization, one example of which is mentioned in [this](https://blog.grandstack.io/authorization-in-graphql-using-custom-schema-directives-eafa6f5b4658) blog and this is very powerful and declarative. This way, you can specify the scope and add the directive to the respective fields in your SDL and it can do the job for you.
+
+**How do I enable real-time applications like Chat, auto-updates and so on in my application with GraphQL?**
+
+There are some options currently to do this.
+
+- The first would be to use [GraphQL Subscriptions](https://www.howtographql.com/graphql-js/7-subscriptions/) which is part of the spec. You have to register the subscriptions upfront and also have support for Websockets if you want to do this.
+- Another way is to do periodic long-time polling which can work at a small scale keeping your application stateless.
+- An another way is to use [live queries](https://medium.com/open-graphql/graphql-subscriptions-vs-live-queries-e38302c7ab8e)
```

Comment:
A bit of self promotion but I think this talk covers a lot of basics around GraphQL Subscriptions - https://youtu.be/Wi7P39sF2nw

Reply:
Haha. Just watched the video. Nice talk for beginners. Will add it.